nltk/test/unit/test_json2csv_corpus.py

```diff
@@ -9,223 +9,199 @@
 """
 Regression tests for `json2csv()` and `json2csv_entities()` in Twitter
 package.
-
 """
+from pathlib import Path
 
-import os
-import unittest
-from tempfile import TemporaryDirectory
+import pytest
 
 from nltk.corpus import twitter_samples
 from nltk.twitter.common import json2csv, json2csv_entities
 
 
-def are_files_identical(filename1, filename2, debug=False):
-    """
-    Compare two files, ignoring carriage returns.
-    """
-    with open(filename1, "rb") as fileA:
-        with open(filename2, "rb") as fileB:
-            result = True
-            for lineA, lineB in zip(
-                sorted(fileA.readlines()), sorted(fileB.readlines())
-            ):
-                if lineA.strip() != lineB.strip():
-                    if debug:
-                        print(
-                            "Error while comparing files. "
-                            + "First difference at line below."
-                        )
-                        print("=> Output file line: {0}".format(lineA))
-                        print("=> Refer. file line: {0}".format(lineB))
-                    result = False
-                    break
-            return result
-
-
-class TestJSON2CSV(unittest.TestCase):
-    def setUp(self):
-        with open(twitter_samples.abspath("tweets.20150430-223406.json")) as infile:
-            self.infile = [next(infile) for x in range(100)]
-        infile.close()
-        self.msg = "Test and reference files are not the same"
-        self.subdir = os.path.join(os.path.dirname(__file__), 'files')
-
-    def tearDown(self):
-        return
-
-    def test_textoutput(self):
-        ref_fn = os.path.join(self.subdir, 'tweets.20150430-223406.text.csv.ref')
-        with TemporaryDirectory() as tempdir:
-            outfn = os.path.join(tempdir, 'tweets.20150430-223406.text.csv')
-            json2csv(self.infile, outfn, ['text'], gzip_compress=False)
-            self.assertTrue(are_files_identical(outfn, ref_fn), msg=self.msg)
-
-    def test_tweet_metadata(self):
-        ref_fn = os.path.join(self.subdir, 'tweets.20150430-223406.tweet.csv.ref')
-        fields = [
+def files_are_identical(pathA, pathB):
+    """ Compare two files, ignoring carriage returns """
+    f1 = pathA.read_bytes().splitlines()
+    f2 = pathB.read_bytes().splitlines()
+    return f1 == f2
+
+
+subdir = Path(__file__).parent / 'files'
+
+@pytest.fixture
+def infile():
+    with open(twitter_samples.abspath("tweets.20150430-223406.json")) as infile:
+        return [next(infile) for x in range(100)]
+
+
+def test_textoutput(tmp_path, infile):
```

Comment:
used the [tmp_path](https://docs.pytest.org/en/6.2.x/tmpdir.html) fixture instead of TemporaryDirectory